### PR TITLE
Release v3.47.0

### PR DESCRIPTION
**Pull request by kurtmckee**
Added
-----

- Add ``TimersClient.add_app_transfer_data_access_scope`` for ``TimersClient``
  instances which are integrated with ``GlobusApp``. This method registers the
  nested scope dependency for a ``data_access`` requirement for a transfer
  timer. (:pr:`1074`)

- ``SearchQueryV1`` is a new class for submitting complex queries replacing
  the legacy ``SearchQuery`` class. A deprecation warning has been added to the
  ``SearchQuery`` class. (:pr:`1079`)

- Created ``ComputeClientV2`` and ``ComputeClientV3`` classes to support Globus Compute
  API versions 2 and 3, respectively. The canonical ``ComputeClient`` is now a subclass
  of ``ComputeClientV2``, preserving backward compatibility. (:pr:`1096`)

- Added the ``ComputeClientV3.submit()``, ``ComputeClientV2.submit()``,
  ``ComputeClientV2.get_task()``, ``ComputeClientV2.get_task_batch()``,
  and ``ComputeClientV2.get_task_group()`` methods. (:pr:`1094`)

Changed
-------

- Improved error messaging around EOF errors when prompting for code during a command
  line login flow (:pr:`1093`)

Deprecated
----------

- Deprecated the ``ComputeFunctionDocument`` and ``ComputeFunctionMetadata`` classes.
  This change reflects an early design adjustment to better align with the existing
  Globus Compute SDK. (:pr:`1092`)

Development
-----------

- Introduce a ``toxfile.py`` to ensure clean builds during development. (:pr:`1098`)

----
📚 Documentation preview 📚: https://globus-sdk-python--1103.org.readthedocs.build/en/1103/

### CODE REVIEW

**Review by derek-globus — APPROVED** [review]

**Comment by sirosen** [comment]
I just realized, should we add a changelog entry for lazy import changes? I don't know why I didn't write one originally.

**Comment by kurtmckee** [comment]
Yeah, I'll add something about that before merging and wait for your (@sirosen) approval on the PR.

**Review by sirosen — APPROVED** [review]

**Pull request merged**
